### ISSUE TITLE
RT-Extension-BriefHistory-0.01: empty ticket display

### ISSUE THREAD

**Issue by jalbstmeijer** [cloos/RT-Extension-BriefHistory#1]
Hi,

I'm testing RT-Extension-BriefHistory-0.01 on my rt-4.0.10 installation.
I do not get errors, but the History in the Display view becomes empty after enabling this extension. (where create, correspond and comment should still be visible)
Is it possible it is not compatible with rt-4.0.10?
Please let me know if I can supply you any debug information.

Gr, J

**Comment by netsandbox** [comment]
Hi J,

I just installed RT 4.0.10 and my extension and can't see any problems.

Did you see errors in your webserver logs?
Do you have other extensions installed?

Chris

Am 30.07.2014 um 14:37 schrieb jalbstmeijer:

**Comment by jalbstmeijer** [comment]
Hi,

Just tested again.

It is the only plugin enabled;
Set(@Plugins, qw(RT::Extension::BriefHistory));

Both in the regular view as in Selfservice displaying a ticket returns an empty view, see screenshot.

![rt](https://cloud.githubusercontent.com/assets/362000/4062957/e0a83566-2e03-11e4-92fd-3ccd3f303784.png)

No RT errors in messages and no Apache errors.

**Comment by jalbstmeijer** [comment]
maybe interesting to know.
even when commenting the  Set(@Plugins,  line, clearing cache and restarting apache.
The output stays empty.
Only after I remove /usr/share/rt4/html/Callbacks/BriefHistory/Ticket/Elements/ShowHistory/SkipTransaction I get my output back.

**Comment by netsandbox** [comment]
Am 27.08.2014 um 18:17 schrieb jalbstmeijer:

> maybe interesting to know.
> even when commenting the Set(@Plugins, line, clearing cache and
> restarting apache.
> The output stays empty.
> Only after I remove
> /usr/share/rt4/html/Callbacks/BriefHistory/Ticket/Elements/ShowHistory/SkipTransaction
> I get my output back.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/cloos/RT-Extension-BriefHistory/issues/1#issuecomment-53599175.

Hi,

/usr/share/rt4/html/Callbacks/...
is the wrong installation path for the plugin files.
Which OS do you use?
How do you installed my plugin? From CPAN or from Github?

You can lookup the correct plugin path within the RT WebUI under
Tools->Configuration->Tools->System Configuration
Search on this page for "RT::LocalPluginPath"

If you run "make install" my plugin should install under the
"RT::LocalPluginPath" directory.

Chris

**Comment by jalbstmeijer** [comment]
Hi,

> /usr/share/rt4/html/Callbacks/... is the wrong installation path for the plugin files.

See install transcript below. Nothing manual done by me choosing this location.

> Which OS do you use?

Centos 6.

> How do you installed my plugin? From CPAN or from Github?

Downloaded from Github,

wget https://github.com/cloos/RT-Extension-BriefHistory/archive/0.01.tar.gz

untarred and done the following:

root@tickets RT-Extension-BriefHistory-0.01]# perl Makefile.PL 
Using RT configuration from /usr/share/perl5/vendor_perl/RT.pm:
./etc => /usr/local/etc/rt4/Extension-BriefHistory
./html => /usr/share/rt4/html
./lib => /usr/local/lib/rt4/lib
Checking if your kit is complete...
Warning: the following files are missing in your kit:
Makefile
Please inform the author.
Writing Makefile for RT::Extension::BriefHistory

[root@tickets RT-Extension-BriefHistory-0.01]# make
cp lib/RT/Extension/BriefHistory.pm blib/lib/RT/Extension/BriefHistory.pm
Manifying blib/man3/RT::Extension::BriefHistory.3pm

[root@tickets RT-Extension-BriefHistory-0.01]# make install
Installing /usr/local/lib/rt4/lib/RT/Extension/BriefHistory.pm
Installing /usr/local/lib/rt4/man/man3/RT::Extension::BriefHistory.3pm
Appending installation info to /usr/local/lib/rt4/lib/perllocal.pod
Installing /usr/local/etc/rt4/Extension-BriefHistory/BriefHistory_Config.pm
Installing /usr/share/rt4/html/Callbacks/BriefHistory/Ticket/Elements/ShowHistory/SkipTransaction

> You can lookup the correct plugin path within the RT WebUI under 
> Tools->Configuration->Tools->System Configuration Search on this page for 
> "RT::LocalPluginPath" If you run "make install" my plugin should install under the 
> "RT::LocalPluginPath" directory.

I see no "RT::LocalPluginPath" on that page, only:

RT::BasePath    /usr
RT::BinPath     /usr/sbin
RT::EtcPath     /etc/rt4
RT::LexiconPath     /usr/share/rt4/po
RT::LocalEtcPath    /usr/local/etc/rt4
RT::LocalLexiconPath    /usr/local/lib/rt4/po
RT::LocalLibPath    /usr/local/lib/rt4/lib
RT::LocalPath   /usr/local/lib/rt4
RT::MasonComponentRoot  /usr/share/rt4/html
RT::MasonDataDir    /var/cache/rt4/mason_data
RT::MasonLocalComponentRoot     /usr/local/lib/rt4/html
RT::MasonSessionDir     /var/cache/rt4/session_data
RT::SbinPath    /usr/sbin
RT::VERSION     4.0.10
RT::VarPath     /var/lib/rt4 

Gr, J

**Comment by netsandbox** [comment]
Am 29.08.2014 um 12:59 schrieb jalbstmeijer:

> Hi,
> 
> ```
> /usr/share/rt4/html/Callbacks/... is the wrong installation path for
> the plugin files.
> ```
> 
> See install transcript below. Nothing manual done by me choosing this
> location.
> 
> ```
> Which OS do you use?
> ```
> 
> Centos 6.
> 
> ```
> How do you installed my plugin? From CPAN or from Github?
> ```
> 
> Downloaded from Github,
> 
> wget https://github.com/cloos/RT-Extension-BriefHistory/archive/0.01.tar.gz
> 
> untarred and done the following:
> 
> root@tickets RT-Extension-BriefHistory-0.01]# perl Makefile.PL
> Using RT configuration from /usr/share/perl5/vendor_perl/RT.pm:
> ./etc => /usr/local/etc/rt4/Extension-BriefHistory
> ./html => /usr/share/rt4/html
> ./lib => /usr/local/lib/rt4/lib
> Checking if your kit is complete...
> Warning: the following files are missing in your kit:
> Makefile
> Please inform the author.
> Writing Makefile for RT::Extension::BriefHistory
> 
> [root@tickets RT-Extension-BriefHistory-0.01]# make
> cp lib/RT/Extension/BriefHistory.pm blib/lib/RT/Extension/BriefHistory.pm
> Manifying blib/man3/RT::Extension::BriefHistory.3pm
> 
> [root@tickets RT-Extension-BriefHistory-0.01]# make install
> Installing /usr/local/lib/rt4/lib/RT/Extension/BriefHistory.pm
> Installing /usr/local/lib/rt4/man/man3/RT::Extension::BriefHistory.3pm
> Appending installation info to /usr/local/lib/rt4/lib/perllocal.pod
> Installing /usr/local/etc/rt4/Extension-BriefHistory/BriefHistory_Config.pm
> Installing
> /usr/share/rt4/html/Callbacks/BriefHistory/Ticket/Elements/ShowHistory/SkipTransaction
> 
> ```
> You can lookup the correct plugin path within the RT WebUI under
> Tools->Configuration->Tools->System Configuration Search on this
> page for
> "RT::LocalPluginPath" If you run "make install" my plugin should
> install under the
> "RT::LocalPluginPath" directory.
> ```
> 
> I see no "RT::LocalPluginPath" on that page, only:
> 
> RT::BasePath /usr
> RT::BinPath /usr/sbin
> RT::EtcPath /etc/rt4
> RT::LexiconPath /usr/share/rt4/po
> RT::LocalEtcPath /usr/local/etc/rt4
> RT::LocalLexiconPath /usr/local/lib/rt4/po
> RT::LocalLibPath /usr/local/lib/rt4/lib
> RT::LocalPath /usr/local/lib/rt4
> RT::MasonComponentRoot /usr/share/rt4/html
> RT::MasonDataDir /var/cache/rt4/mason_data
> RT::MasonLocalComponentRoot /usr/local/lib/rt4/html
> RT::MasonSessionDir /var/cache/rt4/session_data
> RT::SbinPath /usr/sbin
> RT::VERSION 4.0.10
> RT::VarPath /var/lib/rt4
> 
> Gr, J
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/cloos/RT-Extension-BriefHistory/issues/1#issuecomment-53862500.

Hi J,

I assume you installed RT from the Centos Package and not from source?
I think there is something wrong with the package.
There must be an "RT::LocalPluginPath" defined as all RT extensions use
this as installation root.

Sadly I'm not familiar with Centos and don't have an Centos test system
available. I send a message to the rt users list to hopefully get some
response that will help you:
http://www.gossamer-threads.com/lists/rt/users/125031

Chris

**Comment by jalbstmeijer** [comment]
Hi,

> I assume you installed RT from the Centos Package and not from source?

Yes to be exact http://springdale.math.ias.edu/data/puias/unsupported/6/x86_64/rt4-4.0.10-2.sdl6.noarch.rpm

btw, I also tested installing an other extension, which seems to be working fine.
This says nothing about  you  being correct or not, just observing an other extension not depending on RT::LocalPluginPath.

[root@tickets RT-Extension-HideWidgets]# perl Makefile.PL 
Cannot determine perl version info from lib/RT/Extension/HideWidgets.pm
Using RT configuration from /usr/share/perl5/vendor_perl/RT.pm:
./html => /usr/share/rt4/html
./lib => /usr/local/lib/rt4/lib
Cannot determine perl version info from lib/RT/Extension/HideWidgets.pm
Checking if your kit is complete...
Looks good
Writing Makefile for RT::Extension::HideWidgets

[root@tickets RT-Extension-HideWidgets]# make
cp lib/RT/Extension/HideWidgets.pm blib/lib/RT/Extension/HideWidgets.pm
Manifying blib/man3/RT::Extension::HideWidgets.3pm

[root@tickets RT-Extension-HideWidgets]# make install
Installing /usr/local/lib/rt4/lib/RT/Extension/HideWidgets.pm
Installing /usr/local/lib/rt4/man/man3/RT::Extension::HideWidgets.3pm
Appending installation info to /usr/local/lib/rt4/lib/perllocal.pod
Installing /usr/share/rt4/html/Ticket/Elements/ShowSummary

**Comment by netsandbox** [comment]
Am 30.08.2014 um 22:56 schrieb jalbstmeijer:

> [root@tickets RT-Extension-HideWidgets]# perl Makefile.PL
> Cannot determine perl version info from lib/RT/Extension/HideWidgets.pm
> Using RT configuration from /usr/share/perl5/vendor_perl/RT.pm:
> ./html => /usr/share/rt4/html
> ./lib => /usr/local/lib/rt4/lib
> Cannot determine perl version info from lib/RT/Extension/HideWidgets.pm
> Checking if your kit is complete...
> Looks good
> Writing Makefile for RT::Extension::HideWidgets
> 
> [root@tickets RT-Extension-HideWidgets]# make
> cp lib/RT/Extension/HideWidgets.pm blib/lib/RT/Extension/HideWidgets.pm
> Manifying blib/man3/RT::Extension::HideWidgets.3pm
> 
> [root@tickets RT-Extension-HideWidgets]# make install
> Installing /usr/local/lib/rt4/lib/RT/Extension/HideWidgets.pm
> Installing /usr/local/lib/rt4/man/man3/RT::Extension::HideWidgets.3pm
> Appending installation info to /usr/local/lib/rt4/lib/perllocal.pod
> Installing /usr/share/rt4/html/Ticket/Elements/ShowSummary

The last line actually mean that the core RT file ShowSummary was
overwritten by the extension version, which is not how extensions in RT
work. One of the RT developers also confirmed that you RT package is
somewhat broken. You can view this in the link I send you in my previous
message.
Can you please send me you RT/Generated.pm file, the developer want to
have a look at the defined vars in this file. I assume it should be
under /usr/share/perl5/vendor_perl/.

Chris

**Comment by jalbstmeijer** [comment]
Hi Chris,

Thank for looking in to this!

Here my /usr/share/perl5/vendor_perl/RT/Generated.pm

# BEGIN BPS TAGGED BLOCK {{{

#

# COPYRIGHT:

#

# This software is Copyright (c) 1996-2013 Best Practical Solutions, LLC

# sales@bestpractical.com

#

# (Except where explicitly superseded by other copyright notices)

#
#

# LICENSE:

#

# This work is made available to you under the terms of Version 2 of

# the GNU General Public License. A copy of that license should have

# been provided with this software, but in any event can be snarfed

# from www.gnu.org.

#

# This work is distributed in the hope that it will be useful, but

# WITHOUT ANY WARRANTY; without even the implied warranty of

# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU

# General Public License for more details.

#

# You should have received a copy of the GNU General Public License

# along with this program; if not, write to the Free Software

# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA

# 02110-1301 or visit their web page on the internet at

# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html.

#
#

# CONTRIBUTION SUBMISSION POLICY:

#

# (The following paragraph is not intended to limit the rights granted

# to you to modify and distribute this software under the terms of

# the GNU General Public License and is only of importance to you if

# you choose to contribute your changes and enhancements to the

# community by submitting them to Best Practical Solutions, LLC.)

#

# By intentionally submitting any modifications, corrections or

# derivatives to this work, or any other work intended for use with

# Request Tracker, to Best Practical Solutions, LLC, you confirm that

# you are the copyright holder for those contributions and you grant

# Best Practical Solutions,  LLC a nonexclusive, worldwide, irrevocable,

# royalty-free, perpetual, license to use, copy, create derivative

# works based on those contributions, and sublicense and distribute

# those contributions and any derivatives thereof.

#

# END BPS TAGGED BLOCK }}}

package RT;
use warnings;
use strict;

our $VERSION = '4.0.10';

$BasePath = '/usr';
$EtcPath = '/etc/rt4';
$BinPath = '/usr/sbin';
$SbinPath = '/usr/sbin';
$VarPath = '/var/lib/rt4';
$LexiconPath = '/usr/share/rt4/po';
$PluginPath = '';
$LocalPath = '/usr/local/lib/rt4';
$LocalEtcPath = '/usr/local/etc/rt4';
$LocalLibPath        =    '/usr/local/lib/rt4/lib';
$LocalLexiconPath = '/usr/local/lib/rt4/po';
$LocalPluginPath = '';

# $MasonComponentRoot is where your rt instance keeps its mason html files

$MasonComponentRoot = '/usr/share/rt4/html';

# $MasonLocalComponentRoot is where your rt instance keeps its site-local

# mason html files.

$MasonLocalComponentRoot = '/usr/local/lib/rt4/html';

# $MasonDataDir Where mason keeps its datafiles

$MasonDataDir = '/var/cache/rt4/mason_data';

# RT needs to put session data (for preserving state between connections

# via the web interface)

$MasonSessionDir = '/var/cache/rt4/session_data';

1;

**Comment by jalbstmeijer** [comment]
Based on the RT/Generated.pm remark I decided to fill in the empty vars and this made the extension work for me. Thanks.

--- /root/rt-orig/usr/share/perl5/vendor_perl/RT/Generated.pm   2014-10-11 20:21:55.103769248 +0200
+++ /usr/share/perl5/vendor_perl/RT/Generated.pm    2014-10-11 20:22:13.408072475 +0200
@@ -60,12 +60,12 @@
 $SbinPath = '/usr/sbin';
 $VarPath = '/var/lib/rt4';
 $LexiconPath = '/usr/share/rt4/po';
-$PluginPath = '';
+$PluginPath = '/usr/lib/rt4/plugins';
 $LocalPath = '/usr/local/lib/rt4';
 $LocalEtcPath = '/usr/local/etc/rt4';
 $LocalLibPath        =    '/usr/local/lib/rt4/lib';
 $LocalLexiconPath = '/usr/local/lib/rt4/po';
-$LocalPluginPath = '';
+$LocalPluginPath = '/usr/local/lib/rt4/plugins';
 # $MasonComponentRoot is where your rt instance keeps its mason html files
 $MasonComponentRoot = '/usr/share/rt4/html';
 # $MasonLocalComponentRoot is where your rt instance keeps its site-local
